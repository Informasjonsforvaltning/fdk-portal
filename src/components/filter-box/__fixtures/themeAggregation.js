export const themeAggregation = {
  buckets: [
    { key: 'GOVE', count: 104 },
    { key: 'EDUC', count: 11 },
    { key: 'HEAL', count: 8 },
    { key: 'Ukjent', count: 7 },
    { key: 'AGRI', count: 1 },
    { key: 'TEST', count: 1 }
  ]
};
