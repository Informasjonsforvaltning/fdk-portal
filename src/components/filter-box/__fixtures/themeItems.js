export const themeItems = {
  AGRI: {
    id: 'http://publications.europa.eu/resource/authority/data-theme/AGRI',
    code: 'AGRI',
    startUse: '2015-10-01',
    title: {
      pt: 'Agricultura, pesca, silvicultura e alimentação',
      bg: 'Селско стопанство, рибарство, горско стопанство и храни'
    },
    conceptSchema: {
      id: 'http://publications.europa.eu/resource/authority/data-theme',
      title: { en: 'Dataset types Named Authority List' }
    }
  }
};
