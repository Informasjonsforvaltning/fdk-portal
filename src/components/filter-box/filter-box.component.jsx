import React from 'react';
import PropTypes from 'prop-types';
import { Collapse } from 'reactstrap';
import cx from 'classnames';
import _ from 'lodash';

import localization from '../../lib/localization';
import { FilterOption } from '../filter-option/filter-option.component';
import './filter-box.scss';

export class FilterBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      openFilter: true,
      open: false
    };
    this.toggleFilter = this.toggleFilter.bind(this);
    this.toggleList = this.toggleList.bind(this);
  }

  toggleFilter() {
    const { openFilter } = this.state;
    this.setState({ openFilter: !openFilter });
  }

  toggleList() {
    const { open } = this.state;
    this.setState({ open: !open });
  }

  _renderOptions({ buckets }, onClick, activeFilter) {
    const { open } = this.state;
    const { htmlKey, themesItems } = this.props;
    let filters;
    if (activeFilter) {
      filters = activeFilter.split(',');
    }
    const options = items =>
      items.map((item, index) => {
        // generate unique key, this is used by FilterOption on label htmlFor
        let itemKey = 0;
        if (htmlKey) {
          itemKey = Number.parseInt(`${htmlKey}${index}`, 10);
        }
        let active = false;
        if (filters && filters.includes(item.key)) {
          active = true;
        }
        return (
          <FilterOption
            key={itemKey}
            itemKey={itemKey}
            value={item.key}
            label={item.key}
            count={item.count}
            onClick={onClick}
            active={active}
            themesItems={themesItems}
          />
        );
      });

    if (buckets) {
      const bucketsLength = buckets.length;
      const openIconClass = cx('fa', 'mr-2', {
        'fa-angle-double-down': !open,
        'fa-angle-double-up': open
      });
      return (
        <div>
          {options(buckets.slice(0, 5))}
          {bucketsLength > 5 && (
            <div>
              <Collapse isOpen={open}>
                <div>{options(buckets.slice(5))}</div>
              </Collapse>
              <button
                type="button"
                className="fdk-toggleList"
                onClick={this.toggleList}
              >
                <i className={openIconClass} />
                {open
                  ? localization.facet.showfewer
                  : localization.facet.showmore}
              </button>
            </div>
          )}
        </div>
      );
    }
    return null;
  }

  render() {
    const { openFilter } = this.state;
    const { title, filter, onClick, activeFilter } = this.props;
    const collapseIconClass = cx('fa', {
      'fa-angle-down': !openFilter,
      'fa-angle-up': openFilter
    });
    if (_.get(filter, 'buckets', []).length > 0) {
      return (
        <div className="fdk-panel--filter">
          <div className="fdk-panel__header">
            <button
              type="button"
              className="fdk-toggleFilter p-0 d-flex justify-content-between align-items-center w-100"
              onClick={this.toggleFilter}
            >
              <span>{title}</span>
              <i className={collapseIconClass} />
            </button>
          </div>
          <Collapse isOpen={openFilter}>
            <div className="fdk-panel__content">
              <div className="fdk-items-list">
                {this._renderOptions(filter, onClick, activeFilter)}
              </div>
            </div>
          </Collapse>
        </div>
      );
    }
    return null;
  }
}

FilterBox.defaultProps = {
  htmlKey: null,
  title: null,
  activeFilter: null,
  themesItems: null
};

FilterBox.propTypes = {
  htmlKey: PropTypes.number,
  title: PropTypes.string,
  filter: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  activeFilter: PropTypes.string,
  themesItems: PropTypes.object
};

// export default FilterBox;
