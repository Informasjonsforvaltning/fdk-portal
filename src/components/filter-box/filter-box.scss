@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-panel--filter, .fdk-filter-tree {
  @include themify($themes) {
    background-color: themed('color-neutral-light');
  }
  margin-bottom: .5em;
  border-radius: 5px;
  .fdk-panel__header {
    @include themify($themes) {
      border-bottom: 1px solid themed('color-neutral-lightest');
    }
    line-height: 2em;
    margin-bottom: .5em;
    padding: 1em 1.5em;
    button {
      border: none;
      background-color: transparent;
      cursor: pointer;
      font-size: 1.875rem;
      font-weight: 500;
      i {
        @include themify($themes) {
          color: themed('color-text-link');
        }
        font-size: 2rem;
      }
    }
  }
  .fdk-panel__content {
    font-size: 1.5rem;
    padding: .5em 1.5em;

    .fdk-items-list {
      button {
        background: transparent;
        border: none;
        cursor: pointer;
        padding: 0;
      }
      .fdk-toggleList {
        @include themify($themes) {
          color: themed('color-text-link');
        }
      }
    }
  }
}
