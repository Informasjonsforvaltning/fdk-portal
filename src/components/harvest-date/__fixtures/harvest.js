export const harvest = {
  firstHarvested: '2018-06-25T19":51:05+0200',
  lastHarvested: '2018-07-09T14:40:37+0200',
  lastChanged: '2018-07-09T14:40:37+0200',
  changed: [
    '2018-06-25T19:51:05+0200',
    '2018-06-25T19:51:12+0200',
    '2018-06-26T09:16:46+0200',
    '2018-06-28T09:22:41+0200',
    '2018-06-28T09:52:24+0200',
    '2018-06-29T15:25:53+0200',
    '2018-07-03T10:59:38+0200',
    '2018-07-09T14:40:37+0200'
  ]
};
