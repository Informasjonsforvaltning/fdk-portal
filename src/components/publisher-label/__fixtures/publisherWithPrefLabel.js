export const publisherWithPrefLabel = {
  type: 'no.dcat.datastore.domain.dcat.Publisher',
  valid: false,
  uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
  id: '910244132',
  name: 'RAMSUND OG ROGNAN REVISJON',
  orgPath: '/ANNET/910244132',
  prefLabel: {
    no: 'Ramsund og Rognan revisjon'
  }
};
