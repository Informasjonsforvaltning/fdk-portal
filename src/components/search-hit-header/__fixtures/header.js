export const header = {
  title: 'Test tittel',
  publisherLabel: 'Eies av',
  publisher: {
    id: '974760673'
  }
};
