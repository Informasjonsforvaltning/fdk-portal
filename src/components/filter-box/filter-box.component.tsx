import React, { ChangeEvent } from 'react';
import { Collapse } from 'reactstrap';
import _ from 'lodash';
import Select from 'react-select';

import localization from '../../lib/localization';
import { FilterOption } from '../filter-option/filter-option.component';
import './filter-box.scss';

import CollapseTextIcon from '../../img/icon-collapse-text-sm.svg';
import ExpandTextIcon from '../../img/icon-expand-text-sm.svg';
import { FilterSearchOption } from '../../types';

interface Props {
  htmlKey?: number;
  title?: string;
  filter: Record<string, any>;
  groupByPrefix?: string[];
  searchable?: boolean;
  onClick: (change: FilterChange) => void;
  activeFilter?: any;
  referenceDataItems?: Record<string, any>;
  filters?: Record<string, any>;
  capitalizeOption?: boolean;
}

export interface FilterChange {
  value: string;
  checked: boolean;
}

interface State {
  openFilter: boolean;
  open: boolean;
  filterSearchValue: any;
}

type BucketItem = {
  label?: string;
  key: string;
  count: number;
};

const DEFAULT_GROUP = 'default';

export class FilterBox extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      openFilter: true,
      open: false,
      filterSearchValue: null
    };
    this.toggleFilter = this.toggleFilter.bind(this);
    this.toggleList = this.toggleList.bind(this);
  }

  toggleFilter() {
    const { openFilter } = this.state;
    this.setState({ openFilter: !openFilter });
  }

  toggleList() {
    const { open } = this.state;
    this.setState({ open: !open });
  }

  _getGroup(item: BucketItem, prefixes: string[]): string {
    if (prefixes.length > 0) {
      return item.key.startsWith(prefixes[0])
        ? prefixes[0]
        : this._getGroup(item, prefixes.slice(1));
    }
    return DEFAULT_GROUP;
  }

  _renderOptions(
    filter: any,
    onClick: any,
    activeFilter: any,
    allFilters: any,
    groupByPrefix: string[],
    capitalizeOption: any
  ) {
    const { open } = this.state;
    const { htmlKey, referenceDataItems } = this.props;
    const filters: any[] = [];
    if (activeFilter) {
      filters.push(...activeFilter.split(','));
    }
    if (allFilters && 'opendata' in allFilters) {
      filters.push('OPEN_DATA');
    }

    const groupByPrefixes = (data: BucketItem[]) =>
      data.reduce((results: Record<string, BucketItem[]>, item) => {
        const group = this._getGroup(item, groupByPrefix);
        const label = item.key.replace(new RegExp(`${group}\\s?`), '');
        item.label = label || localization.facet.formatType.UNKNOWN;
        results[group] = results[group] || [];
        results[group].push(item);
        return results;
      }, {});

    const options = (items: BucketItem[], groupIndex: number) =>
      items.map((item, index) => {
        // generate unique key, this is used by FilterOption on label htmlFor
        let itemKey = 0;
        if (htmlKey) {
          itemKey = Number.parseInt(`${htmlKey}${groupIndex}${index}`, 10);
        }
        let active = false;
        if (filters.includes(item.key)) {
          active = true;
        }

        return (
          <FilterOption
            key={itemKey}
            itemKey={itemKey}
            value={item.key}
            label={item.label ?? item.key}
            count={item.count}
            onClick={onClick}
            active={active}
            capitalize={capitalizeOption}
            referenceDataItems={referenceDataItems}
          />
        );
      });

    if (filter?.buckets) {
      return Object.entries(groupByPrefixes(filter.buckets))
        .filter(
          ([group]) => groupByPrefix.length === 0 || group !== DEFAULT_GROUP
        )
        .map(([group, items], groupIndex, groups) => (
          <div key={`group-${groupIndex}`}>
            {groups.length > 1 && (
              <div
                key={`group-title-${groupIndex}`}
                className='fdk-items-title'
              >
                {localization.facet.formatType[group]}
              </div>
            )}
            <div key={`group-items-${groupIndex}`}>
              {options(items.slice(0, 5), groupIndex)}
              {items.length > 5 && (
                <div>
                  <Collapse isOpen={open}>
                    <div>{options(items.slice(5), groupIndex)}</div>
                  </Collapse>
                  <button
                    type='button'
                    className='fdk-toggleList'
                    onClick={this.toggleList}
                  >
                    <img
                      src={open ? CollapseTextIcon : ExpandTextIcon}
                      alt=''
                      className='mr-2'
                    />
                    {open
                      ? localization.facet.showfewer
                      : localization.facet.showmore}
                  </button>
                </div>
              )}
            </div>
          </div>
        ));
    }
    return null;
  }

  render() {
    const { openFilter, filterSearchValue } = this.state;
    const {
      title,
      filter,
      searchable,
      onClick,
      activeFilter,
      filters,
      groupByPrefix = [],
      capitalizeOption = true
    } = this.props;

    const filterSearchOptions = filter?.buckets
      ?.map((item: BucketItem) => {
        const group = this._getGroup(item, groupByPrefix);
        const label = item.key.replace(new RegExp(`${group}\\s?`), '');
        return {
          value: item.key,
          label: label || localization.facet.formatType.UNKNOWN
        };
      })
      .sort((a: FilterSearchOption, b: FilterSearchOption) =>
        a.label.localeCompare(b.label)
      );

    const handleOnClick = ({
      target: { value, checked }
    }: ChangeEvent<HTMLInputElement>) => {
      if (onClick) {
        onClick({ value, checked });
        this.setState({ filterSearchValue: null });
      }
    };

    if (_.get(filter, 'buckets', []).length > 0) {
      return (
        <div className='fdk-panel--filter'>
          <div className='fdk-panel__header'>
            <button
              type='button'
              className='fdk-toggleFilter p-0 d-flex justify-content-between align-items-center w-100'
              onClick={this.toggleFilter}
            >
              <span>{title}</span>
            </button>
          </div>
          <Collapse isOpen={openFilter}>
            <div className='fdk-panel__content'>
              {searchable && (
                <div className='fdk-filter-search'>
                  <Select
                    aria-label={`${
                      localization.facet.searchFor
                    } ${title?.toLowerCase()}`}
                    options={filterSearchOptions}
                    value={filterSearchValue}
                    onChange={({ value }: any) => {
                      onClick({ value, checked: true });
                      this.setState({ filterSearchValue: null });
                    }}
                    placeholder={`${
                      localization.facet.searchFor
                    } ${title?.toLowerCase()}`}
                    searchPromptText={localization.report.typeToSearch}
                    backspaceRemoves
                    theme={theme => ({
                      ...theme,
                      colors: {
                        ...theme.colors,
                        neutral50: '#666'
                      }
                    })}
                  />
                </div>
              )}
              <div className='fdk-items-list'>
                {this._renderOptions(
                  filter,
                  handleOnClick,
                  activeFilter,
                  filters,
                  groupByPrefix,
                  capitalizeOption
                )}
              </div>
            </div>
          </Collapse>
        </div>
      );
    }
    return null;
  }
}
