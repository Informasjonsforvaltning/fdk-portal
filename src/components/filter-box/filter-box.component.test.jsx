import React from 'react';
import { shallow } from 'enzyme';
import { FilterBox } from './filter-box.component';
import { themeAggregation } from './__fixtures/themeAggregation';
import { themeItems } from './__fixtures/themeItems';

let onClick;
let defaultProps;
let wrapper;

beforeEach(() => {
  onClick = jest.fn();
  defaultProps = {
    title: 'Testbox',
    filter: themeAggregation,
    onClick,
    activeFilter: null,
    themesItems: themeItems
  };
  wrapper = shallow(<FilterBox {...defaultProps} />);
});

test('should render FilterBox correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render FilterBox correctly with activeFilter', () => {
  wrapper.setProps({
    activeFilter: 'GOVE'
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render FilterBox correctly with empty filter', () => {
  wrapper.setProps({
    filter: {}
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle toggleFilter', () => {
  wrapper.find('button.fdk-toggleFilter').simulate('click');
  expect(wrapper.state('openFilter')).toBe(false);
});

test('should handle toggleList', () => {
  wrapper.find('button.fdk-toggleList').simulate('click');
  expect(wrapper.state('open')).toBe(true);
});
