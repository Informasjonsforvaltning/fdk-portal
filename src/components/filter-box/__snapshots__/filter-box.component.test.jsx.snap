// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterBox correctly 1`] = `
<div
  className="fdk-panel--filter"
>
  <div
    className="fdk-panel__header"
  >
    <button
      className="fdk-toggleFilter p-0 d-flex justify-content-between align-items-center w-100"
      onClick={[Function]}
      type="button"
    >
      <span>
        Testbox
      </span>
      <i
        className="fa fa-angle-up"
      />
    </button>
  </div>
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <div
      className="fdk-panel__content"
    >
      <div
        className="fdk-items-list"
      >
        <div>
          <FilterOption
            active={false}
            count={104}
            displayClass={null}
            itemKey={0}
            key="0"
            label="GOVE"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="GOVE"
          />
          <FilterOption
            active={false}
            count={11}
            displayClass={null}
            itemKey={0}
            key="0"
            label="EDUC"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="EDUC"
          />
          <FilterOption
            active={false}
            count={8}
            displayClass={null}
            itemKey={0}
            key="0"
            label="HEAL"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="HEAL"
          />
          <FilterOption
            active={false}
            count={7}
            displayClass={null}
            itemKey={0}
            key="0"
            label="Ukjent"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="Ukjent"
          />
          <FilterOption
            active={false}
            count={1}
            displayClass={null}
            itemKey={0}
            key="0"
            label="AGRI"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="AGRI"
          />
          <div>
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <div>
                <FilterOption
                  active={false}
                  count={1}
                  displayClass={null}
                  itemKey={0}
                  key="0"
                  label="TEST"
                  labelRaw={null}
                  onClick={[MockFunction]}
                  themesItems={
                    Object {
                      "AGRI": Object {
                        "code": "AGRI",
                        "conceptSchema": Object {
                          "id": "http://publications.europa.eu/resource/authority/data-theme",
                          "title": Object {
                            "en": "Dataset types Named Authority List",
                          },
                        },
                        "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                        "startUse": "2015-10-01",
                        "title": Object {
                          "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                          "pt": "Agricultura, pesca, silvicultura e alimentação",
                        },
                      },
                    }
                  }
                  value="TEST"
                />
              </div>
            </Collapse>
            <button
              className="fdk-toggleList"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa mr-2 fa-angle-double-down"
              />
              Show more
            </button>
          </div>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`should render FilterBox correctly with activeFilter 1`] = `
<div
  className="fdk-panel--filter"
>
  <div
    className="fdk-panel__header"
  >
    <button
      className="fdk-toggleFilter p-0 d-flex justify-content-between align-items-center w-100"
      onClick={[Function]}
      type="button"
    >
      <span>
        Testbox
      </span>
      <i
        className="fa fa-angle-up"
      />
    </button>
  </div>
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <div
      className="fdk-panel__content"
    >
      <div
        className="fdk-items-list"
      >
        <div>
          <FilterOption
            active={true}
            count={104}
            displayClass={null}
            itemKey={0}
            key="0"
            label="GOVE"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="GOVE"
          />
          <FilterOption
            active={false}
            count={11}
            displayClass={null}
            itemKey={0}
            key="0"
            label="EDUC"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="EDUC"
          />
          <FilterOption
            active={false}
            count={8}
            displayClass={null}
            itemKey={0}
            key="0"
            label="HEAL"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="HEAL"
          />
          <FilterOption
            active={false}
            count={7}
            displayClass={null}
            itemKey={0}
            key="0"
            label="Ukjent"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="Ukjent"
          />
          <FilterOption
            active={false}
            count={1}
            displayClass={null}
            itemKey={0}
            key="0"
            label="AGRI"
            labelRaw={null}
            onClick={[MockFunction]}
            themesItems={
              Object {
                "AGRI": Object {
                  "code": "AGRI",
                  "conceptSchema": Object {
                    "id": "http://publications.europa.eu/resource/authority/data-theme",
                    "title": Object {
                      "en": "Dataset types Named Authority List",
                    },
                  },
                  "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                  "startUse": "2015-10-01",
                  "title": Object {
                    "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                    "pt": "Agricultura, pesca, silvicultura e alimentação",
                  },
                },
              }
            }
            value="AGRI"
          />
          <div>
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <div>
                <FilterOption
                  active={false}
                  count={1}
                  displayClass={null}
                  itemKey={0}
                  key="0"
                  label="TEST"
                  labelRaw={null}
                  onClick={[MockFunction]}
                  themesItems={
                    Object {
                      "AGRI": Object {
                        "code": "AGRI",
                        "conceptSchema": Object {
                          "id": "http://publications.europa.eu/resource/authority/data-theme",
                          "title": Object {
                            "en": "Dataset types Named Authority List",
                          },
                        },
                        "id": "http://publications.europa.eu/resource/authority/data-theme/AGRI",
                        "startUse": "2015-10-01",
                        "title": Object {
                          "bg": "Селско стопанство, рибарство, горско стопанство и храни",
                          "pt": "Agricultura, pesca, silvicultura e alimentação",
                        },
                      },
                    }
                  }
                  value="TEST"
                />
              </div>
            </Collapse>
            <button
              className="fdk-toggleList"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa mr-2 fa-angle-double-down"
              />
              Show more
            </button>
          </div>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`should render FilterBox correctly with empty filter 1`] = `""`;
